homeassistant/components/vesync/sensor.py

```diff
@@ -0,0 +1,152 @@
+
+    _async_setup_entities(hass.data[DOMAIN][VS_SENSORS], async_add_entities)
+    return True
+
+
+@callback
+def _async_setup_entities(devices, async_add_entities):
+    """Check if device is online and add entity."""
+    dev_list = []
+    for dev in devices:
+        if DEV_TYPE_TO_HA.get(dev.device_type) == "outlet":
+            dev_list.append(VeSyncPowerSensor(dev))
+            dev_list.append(VeSyncEnergySensor(dev))
+        else:
+            # Not an outlet that supports energy/power, so do not create sensor entities
+            continue
+
+    async_add_entities(dev_list, update_before_add=True)
+
+
+class VeSyncSensorEntity(VeSyncBaseEntity, SensorEntity):
+    """Representation of a sensor describing diagnostics of a VeSync outlet."""
+
+    def __init__(self, plug):
+        """Initialize the VeSync outlet device."""
+        super().__init__(plug)
+        self.smartplug = plug
+
+    @property
+    def entity_category(self):
+        """Return the diagnostic entity category."""
+        return EntityCategory.DIAGNOSTIC
+
+
+class VeSyncPowerSensor(VeSyncSensorEntity):
+    """Representation of current power use for a VeSync outlet."""
+
+    @property
+    def unique_id(self):
+        """Return unique ID for power sensor on device."""
+        return f"{super().unique_id}-power"
+
+    @property
+    def name(self):
+        """Return sensor name."""
+        return f"{super().name} current power"
+
+    @property
+    def device_class(self):
+        """Return the power device class."""
+        return SensorDeviceClass.POWER
+
+    @property
+    def native_value(self):
+        """Return the current power usage in W."""
+        return self.smartplug.power
+
+    @property
+    def native_unit_of_measurement(self):
+        """Return the Watt unit of measurement."""
+        return POWER_WATT
+
+    @property
+    def state_class(self):
+        """Return the measurement state class."""
+        return SensorStateClass.MEASUREMENT
+
+    def update(self):
+        """Update outlet details and energy usage."""
+        self.smartplug.update()
+        self.smartplug.update_energy()
+
+
+class VeSyncEnergySensor(VeSyncSensorEntity):
+    """Representation of current day's energy use for a VeSync outlet."""
+
+    def __init__(self, plug):
+        """Initialize the VeSync outlet device."""
+        super().__init__(plug)
+        self.smartplug = plug
+
+    @property
+    def unique_id(self):
+        """Return unique ID for power sensor on device."""
+        return f"{super().unique_id}-energy"
+
+    @property
+    def name(self):
+        """Return sensor name."""
+        return f"{super().name} energy use today"
+
+    @property
+    def device_class(self):
+        """Return the energy device class."""
+        return SensorDeviceClass.ENERGY
+
+    @property
+    def last_reset(self):
+        """Return datetime representing beginning of day."""
+        return datetime.today()
```

Comment:
I think `last_reset` is not allowed when setting `state_class` to `total_increasing`.

Reply:
Here are our guidelines:
https://developers.home-assistant.io/docs/core/entity/sensor#how-to-choose-state_class-and-last_reset